Examples/Tests/nuclear_fusion/analysis_proton_boron_fusion.py

```diff
@@ -494,9 +494,9 @@ def check_initial_energy1(data, E_com):
 
         assert(np.all(is_close(energy_alpha1_simulation, energy_alpha1_theory, rtol=5.e-8)))
         assert(is_close(np.amax(energy_alpha2_simulation), max_energy_alpha23, rtol=1.e-2))
-        assert(is_close(np.amin(energy_alpha2_simulation), min_energy_alpha23, rtol=1.e-2))
+        assert(is_close(np.amin(energy_alpha2_simulation), min_energy_alpha23, atol=3.218e-15)) # 0.02 MeV abs tol
```

Comment:
Could we change this and the other similar calls to the following instead?
```suggestion
        assert(0 <= np.amin(energy_alpha2_simulation) - min_energy_alpha23 < 3.218e-15) # 0.02 MeV abs tol
```
That way we simultaneously check that the minimum energy is **not** less than the theoretical minimum and that the minimum value is close to the expected minimum. I think it would then also be good to swap the maximum energy checks to the same approach otherwise someone looking at the code in the future might be confused by the asymmetry. What do you think?

Reply:
I made these changes. However, the test now fails.
```
  File "./analysis_proton_boron_fusion.py", line 587, in check_initial_energy2
    assert(0 <= np.amin(energy_alpha2_simulation) - min_energy_alpha23 < 3.218e-14)
```

It appears in `check_initial_energy2` that the difference between the limits. I.e, the `np.amin(energy_alpha2_simulation) - min_energy_alpha23 < 0` and
`np.amax(energy_alpha2_simulation) - max_energy_alpha23 > 0`.

`check_initial_energy1` passes - we are in the center of the momentum frame

`check_initial_energy2` fails - we are in the boron ion rest frame
I assume that there is some small error in the transformation between reference frames causing this.

Reply:
That is unfortunate! Do you know if this same issue was present before changing the reaction pathway (this PR)? If so, we can change the `check_initial_energy2` function back to the way you had it when the test passed (and merge this PR) and create a new issue to look into the origin of the energy exceeding the expected limits.

Reply:
No, I'll have a look later this week and let you know. Yes, no problem if it is an issue in the previous version I'll revert the test and create an issue.

Reply:
Original Version (`check_initial_energy2`). This is what we expect.
<img width="576" alt="image" src="https://github.com/user-attachments/assets/f0a77f8f-b539-479b-952c-0cc024f1dbbe">

My branch:
<img width="576" alt="image" src="https://github.com/user-attachments/assets/368224e5-09a5-4177-8842-b25320cf40af">
For some reason, the minimum value of the 2nd and 3rd alphas generated is less than calculated in the analysis. 

The larger absolute variation is ok because the energy of the 2nd and 3rd alpha particles is also slightly larger via the alpha1 channel.

Reply:
Apologies for the spam. I had a look at the `check_initial_energy1` as well.

Main Branch (FAIL)
<img width="566" alt="image" src="https://github.com/user-attachments/assets/bb7905c7-204a-47cc-ba9f-8f1977a23204">

My branch (PASS)
<img width="566" alt="image" src="https://github.com/user-attachments/assets/cd5e26d0-0828-4399-ba4b-0cf81dbf9f61">


TLDR:
The main dev branch will fail if we check if the alpha particle energy is within the limits we calculate theoretically in the `check_initial_energy1`  function (both max and min are wrong) but pass the `check_initial_energy2`.

My branch will pass `check_initial_energy1` but fails `check_initial_energy2` as some alpha particles are generated with a lower energy than expected. The maximum values are correct.

Could this be some floating point issue that is amplified by the Lorentz transformation that will be performed during the simulation for `check_initial_energy2`? Perhaps this is only apparent as the minimum alpha energies are ~x7 lower in the `alpha1` channel compared to `alpha0` channel.

Summary: In my opinion, this PR is still a net benefit as the alpha spectrum generated matches better what has been observed experimentally (despite the issue above).

If you agree I can revert the tests so that they pass with the updated `E_fusion` and `E_decay` values

Reply:
Thanks for checking @PhysicsDan. I agree with you that since the current `development` version also seem to fail this more rigorous check, albeit in the first test instead of the second, we can merge this to gain the benefit of the more accurate decay channel. So please do revert the check in `check_initial_energy2` to the case where the test passed and let's also summarize the observed discrepancy in a new issue.

Reply:
So just to make sure, the problem is actually only observed with the second case (where the COM frame and labframe are not the same)? If so, the check for energy in case 1 could be as we discussed previously (checking the energy range falls strictly between the theoretical min and max energy) while the second case's check could then be updated once the underlying issue is fixed. 
That change for the case 1 tests is not necessary before merging. I just wanted to point out that case 1 does not have any remaining issues.

Reply:
Yes to confirm `check_initial_energy1` does pass with
```
        assert(0 <= max_energy_alpha23 - np.amax(energy_alpha2_simulation) < 3.218e-15)
        assert(0 <= np.amin(energy_alpha2_simulation) - min_energy_alpha23 < 3.218e-15)
        assert(0 <= max_energy_alpha23 - np.amax(energy_alpha3_simulation) < 3.218e-15)
        assert(0 <= np.amin(energy_alpha3_simulation) - min_energy_alpha23 < 3.218e-15)
```

There are no issues with case 1 in this PR. However, this test would fail with the previous alpha0 channel implementation.